src/akplan/solve.py

```diff
@@ -405,77 +398,66 @@ def export_scheduling_result(
         ValueError: might be raised if the solution of the MILP is infeasible or
             if an AK is not scheduled and allow_unscheduled_aks is False.
     """
-    ak_ids, person_ids, room_ids, timeslot_ids = get_ids(input_data)
-
-    (room_var, time_var, person_var) = dec_vars
+    var_value_dict: dict[str, dict[str, dict[str, int]]] = {
+        "Room": defaultdict(dict),
+        "Time": defaultdict(dict),
+        "Block": defaultdict(dict),
+        "Part": defaultdict(dict),
+    }
 
     def _get_val(var: LpVariable) -> int:
         ret_val = (
-            round(var.solverVar.X)
-            if solved_lp_problem.solver.name == "GUROBI"
+            var.solverVar.X
+            if solved_lp_problem.solver and solved_lp_problem.solver.name == "GUROBI"
             else value(var)
         )
-        return cast(int, ret_val)
+        return cast(int, round(ret_val))
 
-    tmp_res_dir: dict[str, dict[str, dict[str, set[str]]]] = defaultdict(
-        lambda: defaultdict(lambda: defaultdict(set))
-    )
-    tmp_res_dir = defaultdict(lambda: defaultdict(lambda: defaultdict(set)))
-    for ak_id in ak_ids:
-        room_for_ak = None
-        for room_id in room_ids:
-            if _get_val(room_var[ak_id][room_id]) == 1:
-                if room_for_ak is None:
-                    room_for_ak = room_id
-                else:
-                    raise ValueError(f"AK {ak_id} is assigned multiple rooms")
-        if room_for_ak is None:
+    for var in solved_lp_problem.variables():
+        var_cat, idx0, idx1 = var.name.split("_")
+        var_value_dict[var_cat][idx0][idx1] = _get_val(var)
+
+    scheduled_ak_dict: dict[str, dict[str, list[str] | str]] = defaultdict(dict)
+
```

Comment:
```suggestion

    # Handle rooms differntly because we want special handling if AKs are scheduled into more than one room. 
```
We could also decide not to check this in production but only check this in the testbed to simplify this code.